tools/factories.js

```diff
@@ -0,0 +1,159 @@
+        const dob = faker.date.past(70);
+        const Id = arg.Id || randomInt();
+        return Object.freeze({
+            Id,
+            FirstName: firstname,
+            LastName: lastname,
+            DobDay: arg.DobDay || dob.getDate().toString(),
+            DobMonth: arg.DobMonth || (dob.getMonth() + 1).toString(),
+            DobYear: arg.DobYear || dob.getFullYear().toString(),
+            ContactTelephoneNumber: arg.ContactTelephoneNumber || randexp(/020\d{8}/),
+            ContactMobileNumber: arg.ContactMobileNumber || randexp(/07\d{9}/),
+            EmailAddress: arg.EmailAddress || faker.internet.email(firstname, lastname),
+            AddressFirstLine: arg.AddressFirstLine || `Flat ${randexp(/[1-9][A-F]?/)}`,
+            AddressSecondLine: arg.AddressSecondLine || faker.address.streetAddress(),
+            AddressThirdLine:
+                arg.AddressThirdLine || Math.random() > 0.1 ? null : randexp(/Hackney(, London)?/),
+            Postcode: arg.Postcode || randexp(/[A-Z]{1,2}\d ?\d[A-Z]{2}/),
+            Uprn: arg.Uprn || randexp(/1000\d{7}/),
+            Ward: arg.Ward || nullOrValue(faker.random.word()),
+            IsPharmacistAbleToDeliver: arg.IsPharmacistAbleToDeliver || randomNullableBool(),
+            NameAddressPharmacist:
+                arg.NameAddressPharmacist ||
+                nullOrValue(`${faker.company.companyName()}, ${faker.address.streetAddress()}`),
+            GpSurgeryDetails: arg.GpSurgeryDetails || nullOrValue(faker.random.number().toString()),
+            NumberOfChildrenUnder18: arg.NumberOfChildrenUnder18 || nullOrValue(randexp(/[0-5]/)),
+            ConsentToShare: arg.ConsentToShare || randomNullableBool(),
+            RecordStatus: arg.RecordStatus || Math.random() > 0.2 ? 'MASTER' : 'DUPLICATE',
+            NhsNumber: arg.NhsNumber || nullOrValue(randexp(/\d{10}/))
+        });
+    }
+}
+
+class CaseNote_Speculative_V4APIEntity {
+    sample(arg = {}) {
+        const Id = arg.Id || randomInt();
+        return Object.freeze({
+            Id,
+            CaseNote: arg.CaseNote || faker.lorem.words(3),
+            HelpRequestId: arg.HelpRequestId || randomInt(),
+            ResidentId: arg.ResidentId || randomInt(),
+            CreatedAt: arg.CreatedAt || faker.date.recent(40)
+        });
+    }
+}
+
+class CaseNote_FrontEnd_RequestBody {
+    sample(arg = {}) {
+        return Object.freeze({
+            //id, // this is not something in the front-end's control
+            caseNote: arg.caseNote || faker.lorem.words(3),
+            helpRequestId: arg.helpRequestId || randomInt(),
+            residentId: arg.residentId || randomInt()
+            //createdAt: arg.CreatedAt || faker.date.recent(0) // I assume this is set on the back end?
+        });
+    }
+}
+
+module.exports = {
+    ResidentV4APIEntity,
+    HelpRequestV4APIEntity,
+    HelpRequestCallV4APIEntity,
+    CaseNote_Speculative_V4APIEntity,
+    CaseNote_FrontEnd_RequestBody
+};
```

Comment:
Are these only test specific or also prod? 
I feel like they should be separated if it's both?